src/AMRParser.scala

```diff
@@ -77,10 +84,17 @@ scala -classpath . edu.cmu.lti.nlp.amr.AMRParser --stage2-decode -w weights -l l
        val now = System.nanoTime
        val result = a
        val micros = (System.nanoTime - now) / 1000
-       System.err.println("Decoded in %,d microseconds".format(micros))
+       logger(0,"Decoded in %,d microseconds".format(micros))
        result
     }
 
+  // cache the previous Stage2 GraphDecoder object so that it can be re-used. This saves us from having to reload
+  // the weights if JAMR is re-invoked in the same process. Save multiple version using modelName is a key
+    var previousStage2: Map[String, GraphDecoder.Decoder] = Map()
+
+  // an optional callback that lets us get direct access to decoderResultGraph when parsing
+    var resultHandler: Option[Graph => Unit] = None
```

Comment:
I don't see this ever being set to anything other than `None`.

Also, adding side-effecting callbacks makes the code harder to grok/change/debug/test. Is there a way to extract a method for the part of the code that produces `decoderResultGraph` instead?